google_guest_agent/network/systemd_networkd.go

```diff
@@ -0,0 +1,112 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+const configDir = "/etc/systemd/network"
```

Comment:
`configDir` is a very generic term and will certainly clash with other network managers implementation when they come up. Please also consider adding a field to `systemdnetworkd` structure rather than simply using a const.